src/content/blog/fastify.mdx

```diff
@@ -0,0 +1,179 @@
+  and an active community that has been really helpful for support.
+- Solid support for [plugins and encapsulation](https://fastify.dev/docs/latest/Reference/Encapsulation/) that let us structure the application into sub-applications.
+- Support for [validation and serialization](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/) with multiple validators and validation for all inputs and ouputs.
+- [Good support for async/await](https://fastify.dev/docs/latest/Reference/Routes/#async-await), like you might expect from a modern framework.
+- Off-the-shelf integrations for [opentelemetry](https://www.npmjs.com/package/@opentelemetry/instrumentation-fastify) and [Sentry](https://docs.sentry.io/platforms/javascript/guides/fastify/) mean
+  that we could have tracing and telemetry from day one.
+- Finally, [@fastify/express](https://github.com/fastify/fastify-express)
+  let us incrementally migrate from express to fastify.
+- We were already using the [light-my-request](https://github.com/fastify/light-my-request) module to test
+  our express server. light-my-request is written and maintained
+  by the fastify folks, and it is built into fastify.
+
+Of course, fastify is also very fast, but web framework overhead
+wasn't a problem with express and most likely won't be a problem with
+any framework we choose: most of our performance envelope has to
+do with operating [the Val Town runtime][runtimes] and interacting
+with the database.
+
+We considered a few other options, like [Hono](https://hono.dev/) and [Elysia](https://elysiajs.com/).
+Both are lovely projects - we use and recommend Hono a ton
+in Vals. Elysia has convenient typesafety and a lot of features,
+but its support for Node.js is very experimental. And Hono has
+a [first-party OpenAPI plugin](https://github.com/honojs/middleware/tree/main/packages/zod-openapi), but some of the ecosystem around fastify - for tracing,
+incremental migration from express, and plugins - was more built-out.
+
+For simpler projects and especially for situations where we want
+multi-platform compatibility so that code can run on Node.js, Deno,
+or Cloudflare Workers, we reach for Hono.
+
+### The port
+
+By using [@fastify/express](https://github.com/fastify/fastify-express)
+we were able to do this migration entirely incrementally: the first
+fastify PR just added the fastify dependency and wrapped our express
+server with fastify. Then each successive PR moved a few routes
+from express to fastify, until the application has been
+completely ported.
+
+Of course there were some bumps along the road, and lots of details
+to iron out. But as of today we're nearly 100% fastify for Val Town's
+backend, and it's unlocking stability and capabilities across
+the board. Stay tuned for the next update which will dive into those,
+including how this new backend strategy lets us automatically
+generate our new TypeScript SDK!
+
+[openapi]: https://en.wikipedia.org/wiki/OpenAPI_Specification
```

Comment:
Didn't know about this feature of markdown or mdx!

Reply:
Yep, they're called reference-style links - https://www.markdownguide.org/basic-syntax/#reference-style-links